src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -35,15 +35,21 @@
     const ENABLE_MAX_DEPTH = 'enable_max_depth';
     const DEPTH_KEY_PATTERN = 'depth_%s::%s';
     const DISABLE_TYPE_ENFORCEMENT = 'disable_type_enforcement';
-
-    private $propertyTypeExtractor;
-    private $typesCache = array();
-    private $attributesCache = array();
+    const SKIP_NULL_VALUES = 'skip_null_values';
 
     /**
      * @var callable|null
      */
-    private $maxDepthHandler;
+    protected $maxDepthHandler;
+
+    /**
+     * @var bool
+     */
+    protected $skipNullValues = false;
```

Comment:
shouldbe private also (and the typehint looks useless to me, isn't it?)

Reply:
As it's an abstract class, making it private allows to reuse this property in child classes... and we need to do it in API Platform.

Reply:
that's still possible to interact with by decorating the skipNullValues method
protected properties are very hard to deprecate, that's why we really prefer not introducing new ones

Reply:
We need to access to the value too... in the hot path.
But I agree with your point, and it's why need to improve the general design of this component.

As a compromise, can we keep it protected but mark it `@internal`? We'll use it "as is" in API Platform, and if it breaks at some point... we'll upgrade!